packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts

```diff
@@ -0,0 +1,179 @@
+
+  if (permissions === undefined) {
+    context.print.warning('Unable to find layer permissions, falling back to default.');
+    layerConfiguration.permissions = [defaultLayerPermission];
+  } else {
+    layerConfiguration.permissions = [];
+    permissions.map(permission => {
+      switch (permission.type) {
+        case LegacyPermissionEnum.Private:
+          layerConfiguration.permissions.push({ type: PermissionEnum.Private });
+          break;
+        case LegacyPermissionEnum.AwsAccounts:
+          layerConfiguration.permissions.push({ type: PermissionEnum.AwsAccounts, accounts: permission.accounts });
+          break;
+        case LegacyPermissionEnum.AwsOrg:
+          layerConfiguration.permissions.push({ type: PermissionEnum.AwsOrg, orgs: permission.orgs });
+          break;
+        case LegacyPermissionEnum.Public:
+          layerConfiguration.permissions.push({ type: PermissionEnum.Public });
+          break;
+        default:
+          throw new Error('Failed to determine permission type.');
+      }
+    });
+  }
+
+  stateManager.setResourceParametersJson(undefined, categoryName, layerName, {
+    runtimes: runtimecloudTemplateValues,
+    description: '',
+  });
+
+  migrateAmplifyProjectFiles(layerName, 'legacyLayerMigration', context.amplify.getEnvInfo().envName);
+  writeLayerConfigurationFile(layerName, layerConfiguration);
+
+  fs.removeSync(path.join(layerDirPath, LegacyFilename.layerRuntimes));
+  fs.removeSync(path.join(layerDirPath, LegacyFilename.layerParameters));
+  return true;
+}
+
+function isLegacyLayer(layerName: string, layerDirPath: string): LegacyState {
+  if (fs.existsSync(path.join(layerDirPath, LegacyFilename.layerParameters))) {
+    return LegacyState.SINGLE_ENV_LEGACY;
+  }
+
+  if (fs.existsSync(path.join(layerDirPath, LegacyFilename.layerRuntimes))) {
+    return LegacyState.MULTI_ENV_LEGACY;
+  }
+
+  if (fs.existsSync(path.join(layerDirPath, layerConfigurationFileName))) {
+    return LegacyState.NOT_LEGACY;
+  }
+
+  throw new Error(`Lambda layer ${layerName} is missing a state file. Try running "amplify pull --restore". If the issue persists, recreating the layer is the best option. \
+${chalk.red('Ensure your layer content is backed up!')}`);
+}
+
+function migrateAmplifyProjectFiles(layerName: string, latestLegacyHash: string, envName?: string) {
+  const projectRoot = pathManager.findProjectRoot();
+  removeLayerFromTeamProviderInfo(envName, layerName, projectRoot);
+  const meta = stateManager.getMeta(projectRoot);
+
+  if (_.get(meta, [categoryName, layerName, layerVersionMapKey], undefined)) {
+    _.set(meta, [categoryName, layerName, layerVersionMapKey], undefined);
+  }
+
+  _.set(meta, [categoryName, layerName, versionHash], latestLegacyHash);
+  stateManager.setMeta(projectRoot, meta);
+}
+
+export function removeLayerFromTeamProviderInfo(envName: string, layerName: string, projectRoot?: string) {
+  const nonCfnDataKey = 'nonCFNdata';
```

Comment:
naming: `nonCFNData` ?

Reply:
It's correct as is. It's removing the existing key.